fulp_modules/bloodsuckers/bloodsucker/bloodsucker_coffin.dm

```diff
@@ -0,0 +1,176 @@
+/datum/antagonist/bloodsucker/proc/ClaimCoffin(obj/structure/closet/crate/claimed) // NOTE: This can be any "closet" that you are resting AND inside of.
+	// ALREADY CLAIMED
+	if(claimed.resident)
+		if(claimed.resident == owner.current)
+			to_chat(owner, "This is your [src].")
+		else
+			to_chat(owner, "This [src] has already been claimed by another.")
+		return FALSE
+	// Bloodsucker Learns new Recipes!
+	owner.teach_crafting_recipe(/datum/crafting_recipe/vassalrack)
+	owner.teach_crafting_recipe(/datum/crafting_recipe/candelabrum)
+	owner.teach_crafting_recipe(/datum/crafting_recipe/meatcoffin)
+	// This is my Lair
+	coffin = claimed
+	lair = get_area(claimed)
+	// DONE
+	to_chat(owner, "<span class='userdanger'>You have claimed the [claimed] as your place of immortal rest! Your lair is now [lair].</span>")
+	to_chat(owner, "<span class='danger'>You have learned new construction recipes to improve your lair.</span>")
+	to_chat(owner, "<span class='announce'>Bloodsucker Tip: Find new lair recipes in the Misc tab of the <i>Crafting Menu</i>, including the <i>Persuasion Rack</i> for converting crew into Vassals.</span><br><br>")
+	RunLair() // Start
+	return TRUE
+
+// crate.dm
+/obj/structure/closet/crate
+	var/mob/living/resident	// This lets bloodsuckers claim any "closet" as a Coffin, so long as they could get into it and close it. This locks it in place, too.
```

Comment:
except it doesn't, because you're using the crate subtype here, not closets in general